File: CIRCLE/circle.js

// JavaScript implementation to print circle pattern
 
// function to print circle pattern
function printPattern(radius)
{
// dist represents distance to the center
        var dist = parseFloat(0);
 
        // for horizontal movement
        for (var i = 0; i <= 2 * radius; i++)
        { 
          // for vertical movement
          for (var j = 0; j <= 2 * radius; j++)
          {
            dist = Math.sqrt(
              (i - radius) * (i - radius) +
              (j - radius) * (j - radius)
            );
 
            // dist should be in the range (radius - 0.5)
            // and (radius + 0.5) to print stars(*)
            if (dist > radius - 0.5 && dist < radius + 0.5)
            document.write("*");
            else
            document.write("  ");
          }
 
          document.write("<br>");
        }
      }
 
      // Driver Code
       
      var radius = 6;
  
      printPattern(radius);

File: CIRCLE/circle.py
from math import sqrt

def circle(radius):
    pixels = [[' ' for x in range(2*radius+1)] for y in range(2*radius+1)]
    y = radius
    x = 0
    max = (radius + 0.5)**2
    while x <= y:
        pixels[radius+y][radius+x] = '#'
        pixels[radius+y][radius-x] = '#'
        pixels[radius-y][radius+x] = '#'
        pixels[radius-y][radius-x] = '#'
        pixels[radius+x][radius+y] = '#'
        pixels[radius+x][radius-y] = '#'
        pixels[radius-x][radius+y] = '#'
        pixels[radius-x][radius-y] = '#'
        x += 1
        if x*x + y*y >= max:
            y -= 1

    return '\n'.join([''.join([v for v in row]) for row in pixels])

print(circle(11))